index/scorch/snapshot_index.go

```diff
@@ -81,6 +81,8 @@ type IndexSnapshot struct {
 
 	m2        sync.Mutex                                 // Protects the fields that follow.
 	fieldTFRs map[string][]*IndexSnapshotTermFieldReader // keyed by field, recycled TFR's
+
+	updatedFields map[string]*index.UpdateFieldInfo
```

Comment:
Let's add a lot of commentary around here indicating what this is for (for the future), and what each of the attributes of `UpdateFieldInfo` stand for.